pennylane/templates/subroutines/qsvt.py

```diff
@@ -150,6 +150,201 @@ def qsvt(A, angles, wires, convention=None):
     return QSVT(UA, projectors)
 
 
+# pylint: disable=too-many-branches
+def qsvt(A, poly, encoding_wires, block_encoding=None):
+    r"""
+    Implements the Quantum Singular Value Transformation (QSVT) for a matrix or Hamiltonian ``A``, using a polynomial
+    defined by ``poly`` and a block encoding specified by ``block_encoding``.
+
+    .. math::
+
+        \begin{pmatrix}
+        A & * \\
+        * & * \\
+        \end{pmatrix}
+        \xrightarrow{QSVT}
+        \begin{pmatrix}
+        \text{poly}(A) + i \dots & * \\
+        * & * \\
+        \end{pmatrix}
+
+    The polynomial transformation is encoded as the real part of the top left term after applying the operator.
+
+    This function calculates the required phase angles from the polynomial using :func:`~.poly_to_angles`.
+
+    .. note::
+
+        The function ``poly_to_angles``, used within ``qsvt``, is not JIT-compatible, which prevents ``poly`` from being
+        traceable in ``qsvt``. However, ``A`` is traceable and can be optimized by JIT within this function.
+
+    Args:
+
+        A (Union[tensor_like, Operator]): The matrix on which the QSVT will be applied.
+            This can be an array or an object that has a Pauli representation.
+
+        poly (tensor_like): Polynomial coefficients defining the transformation, represented in increasing order of degree.
+            This means the first coefficient corresponds to the constant term, the second to the linear term, and so on.
```

Comment:
Why not just using what we had in poly_to_angles, i.e.,
`coefficients of the polynomial ordered from lowest to highest power`